tasks/main.yml

```
@@ -105,30 +80,32 @@
     state: present
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
+    check_implicit_admin: true
+    login_unix_socket: "{{ mariadb_socket_path }}"
 
-## configuration steps to mimic what 'mysql_secure_installation' does:
+## some configuration steps to mimic what 'mysql_secure_installation' does:
 
 - name: Delete anonymous user
   community.mysql.mysql_user:
     user: ""
-    host: "{{ item }}"
+    host_all: true
     state: absent
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
```

Password may be unset. This would handle this case.

```suggestion
    login_password: "{{ database_root_password | default(omit) }}"
```